base/ltlists.dtx

```diff
@@ -762,12 +824,60 @@
 %
 % \begin{macro}{\if@endpe}
 % \begin{macro}{\@endpefalse}
-% \begin{macro}{\@endpeltrue}
+% \begin{macro}{\@endpetrue}
+% \begin{macro}{\propagate@doendpe}
+%    As outlined above these are no longer simple switches, but we
+%    keep the name because it is used all over the place.
 %    \begin{macrocode}
 \newif\if@endpe
-\@endpefalse
 %    \end{macrocode}
-% \end{macro}\end{macro}\end{macro}
+%    
+% \changes{v1.0u}{2024/06/23}{Set \cs{if@endpe} globally and also set
+%    up migration to the outside}
+%    \begin{macrocode}
+%</2ekernel>
+%<*2ekernel|latexrelease>
+%<latexrelease>\IncludeInRelease{2024/11/01}%
+%<latexrelease>                 {\@endpetrue}{New @endpe handling}%
+%    \end{macrocode}
+%
+%    \begin{macrocode}
+\def\@endpefalse{\global\let\if@endpe\iffalse}
+%    \end{macrocode}
+%
+%    \begin{macrocode}
+\def\@endpetrue {%
+  \global\let\if@endpe\iftrue
+%    \end{macrocode}
+%    If we are inside a group then propagate to the outside:   
+%    \begin{macrocode}
+   \ifnum\currentgrouplevel>\z@
+      \aftergroup\propagate@doendpe
+   \fi
+}
+%    \end{macrocode}
+%    If \cs{if@endpe} is still true after the group ends run \cs{@doendpe}
+%    that in turn runs another \cs{@endpetrue} (besides other things),
+%    thus propagating further, if necessary.
+%    However, if the endpe situation got resolved and it is false
```

Comment:
Need to say what "it" is here, I think.

Reply:
again, I don't see why it is difficult to make the leap from \@endpetrue on the line before to "it is false", but ok :-)

Reply:
Looks like three lines above to me!  
It would probably be OK as "it", but it was not immediately obvious to me (not understanding that there was only one conditional around).